### DISCUSSION TITLE
MEETING: 14 November 2023 - EMu API Integration

### DISCUSSION THREAD

**Discussion by rondlg** [fieldmuseum/EMu-Documentation#934] [MEETINGS]
## Agenda:

- Kick-off discussion around integrating API "stuff" with EMu.
- Can we start with the IUCN Red List API? https://apiv3.iucnredlist.org/

## Notes
How to provide EMu a response that [it can display / can be displayed [somewhere]]?

- [PH] Possible: Take a formatted string and have EMu process/render it. 
- [JJ] - Similar to how the existing tools functionality works?

- [KW] - How do the users (Zach) picture this working client side?
  - Search in EMu + display in EMu, to know what in [Action/other] Catalogue = what threat status

- [JJ] - If we pull data back and store it in EMu how would that be used?
  - Search, filter & create reports

- [JJ] - Reports would go where and do what? 
  - [ZK] As an attribute in a report. To inform management reports and decisions
 
- KW - would that data need to be stored in EMu to search, filter & create reports?
- [JJ] - When you are looking for the status information, does it HAVE to live statically/natively in EMu or not?
  - ZK What are the options?
  - API calls may be run-able during the search / report /etc queries & output can include both EMu catalog + threat status. Need to know where users want to see that output 

- ZK - e.g., search specific species or location, see all EMu catalog from that location + their threatStatus; pass along to park mgrs

- SG - starting with those two (taxon or location], + providing to others – In what form?  [e.g. PDF list of species + statuses]
	&/or non-FM person goes to FM website & can pull that info?
- ZK - sending collated PDF/report’s one example; external users able to search = nice to have; 

- EMu Resources / Registry settings - https://help.emu.axiell.com/v5.1/en/Topics/Common/Resources.htm?Highlight=resources 

## CURL demo - [Recording available](https://github.com/fieldmuseum/EMu-Documentation/assets/702709/f4cf8493-4977-4aa8-b237-952c62ac8c9d)

- Using GBIF API to retrieve data -- e.g.:
  - `curl -i -u gbif-username:pw -H "application/json" -X GET “https://api.gbif.org/v1/species/search?q=Echinus esculentus&datasetKey=19491596-35ae-4a91-9a98-85cf505f1bd3”`

  or 

  - `curl -i -u gbif-username:pw -H "application/json" -X GET “https://api.gbif.org/v1/species/match?genus=Bubalus&species=depressicornis”`

curl -i -u [...] (see curl –help 
-i = 
-u = use username/pw to authenticate CURL request
-H = curl request includes a header, & results will be in specified format 
(e.g. “application/json” = returned data will be in json)
-X = 
-o = specify an output path/file – e.g. “results.json” will write output to a file with that name (in the present working dir.)

200 response = good  (400/500/other errors = check into what went wrong)

- More questions:
  - How does Geolocate work now? Where does data get pulled into now? 
  - Stored readonly?
  - What about history/keeping old data?
  - How often does it get refreshed, by who? Is that automatic when a record is opened or manually? If manually by who.

### Links and Things
- https://data-blog.gbif.org/ 
- https://www.checklistbank.org/ 
- Taxon Desc ext: https://rs.gbif.org/extension/gbif/1.0/description.xml
- Species Distrib ext: http://rs.gbif.org/extension/gbif/1.0/distribution_2022-02-02.xml
- Example record with those extensions, from IUCN redlist dataset: https://www.gbif.org/species/176596775/verbatim 
- JSON to CSV converter. https://www.convertcsv.com/json-to-csv.htm 
- https://docs.gbif.org/course-checklistbank-tutorial/en/explore-the-checklistbank-repository.html 

## Homework
**Deadline/Timeline: v9 in live. Check in before Christmas**
If we take these things from GBIF(See Links and things specifically extensions) and put them together with these other thing and point at the CITES data from the Governmental sites and IUCN (and how they are different)

- Submit stuff to GBIF
KW to check – what does the ‘CITESappendix’ field in the Species Distribution’ extension (http://rs.gbif.org/extension/gbif/1.0/distribution_2022-02-02.xml) cover?

- Submit stuff to AHA that says what we want to do and points to the gbif forum post.

- Use Case: Post Zach presentation to emudata.fieldmuseum.org
  - As pdf with speaker notes
  - Flesh out the use case to include “IN EMu I want to be able to do…”
  - e.g. with v9 screenshots
  - Collate info in github issue

- API call examples
- PH - to check about upgrading FMNH Postman account

**Comment by zkachian** [comment]
Here is a PDF of my presentation at the Axiell User Conference:
[Axiell_Presentation_DangerZone_2023.pdf](https://github.com/fieldmuseum/EMu-Documentation/files/13355360/Axiell_Presentation_DangerZone_2023.pdf)

**Reply by zkachian** [reply]
Should I post this somewhere else as well?

**Reply by rondlg** [reply]
Nope - I'll drop it in the emu-data repo

**Reply by zkachian** [reply]
thanks!

**Reply by rondlg** [reply]
Tis uploaded: https://emudata.fieldmuseum.org/

**Comment by magpiedin** [comment]
FYI - https://discourse.gbif.org/t/ways-to-find-checklist-datasets-with-appendixcites-values-in-a-species-distribution-extension/4203
